### ISSUE TITLE
Solr search engine not updated after adding a user to a group (Humhub 1.2)

### ISSUE THREAD

**Issue by funkycram** [humhub/humhub#3378]
### What steps will reproduce the problem?

Add a user (e.g. id = 39179) into a group (e.g. id=12)  with this code :
```
$user = User::findOne(['id' => 3979]);
Group::findOne(12)->addUser($user);
```

### What is the expected result?

User appear in the group directory : `/directory/members?keyword=&groupId=12`

### What do you get instead?

The user cannot be found in `/directory/members?keyword=&groupId=12`, even a few days later.

### Additional info

The user appear correctly in `/admin/group/manage-group-users?id=12`, which means the user has well been added to the group.

If I edit the user (`/admin/user/edit?id=3979`), the group shows well. If I save the form, without modifying anything (just clicking the “save” button), the user appear immediately in `/directory/members?keyword=&groupId=12`

It looks like my code doesn’t refresh the search engine (SOLR in my case).
Any idea ? Thanks !

This discussion has been started [here in the Community site](https://community.humhub.com/s/module-development/?contentId=173205) but without resolving the problem.

| Q                | A
| ---------------- | ---
| HumHub version   | 1.2
| PHP version      | 7.1
| Operating system | Ubuntu 16.04
| Search engine | Solr

> Note: Please use the [community](https://community.humhub.com) site for any questions on customizing HumHub.
> Note: Also provide the full error stacks from `Administration --> Information --> Logging` and errors from your browser console if possible.

**Comment by acs-ferreira** [comment]
As SOLR is part of EE, you should reach the official support at http://docs.humhub.org/ee-support.html

**Comment by funkycram** [comment]
Thanks @acs-ferreira . Just done, I'll post here the solution.

**Comment by funkycram** [comment]
The bug can be solved in Humhub 1.2 to by adding the following code in `https://github.com/humhub/humhub/blob/v1.2/protected/humhub/modules/user/models/GroupUser.php#L84​` (thanks @luke- )

```
        if ($this->user->isVisible()) {
            Yii::$app->search->update($this->user);
        } else {
            Yii::$app->search->delete($this->user);
        }
```
